auther.go

```diff
@@ -137,12 +129,11 @@ func (a *auther) commonOAuthParams() map[string]string {
 
 // Returns a base64 encoded random 32 byte string.
 func (a *auther) nonce() string {
-	if a.noncer != nil {
-		return a.noncer.Nonce()
+	noncer := DefaultNoncer
```

Comment:
Seems like this defaulting logic would more commonly belong in `newAuther`, retaining the field that was there? 

The test changes in this PR have to do with this choice, and not testing the actual noncer details it seems.

Reply:
That makes sense. I was thinking about the pattern I use when the struct is exported and you're not sure whether it could be nil, but that doesn't apply here.